aptos-move/framework/aptos-stdlib/sources/smart_vector.move

```diff
@@ -0,0 +1,424 @@
+        while (i < 25) {
+            assert!(*borrow(&v1, i) == i, 0);
+            i = i + 1;
+        };
+        destroy(v1);
+    }
+
+    #[test]
+    fun remove_test() {
+        let v = empty();
+        let i = 0u64;
+        while (i < 101) {
+            push_back(&mut v, i);
+            i = i + 1;
+        };
+        let inline_len = vector::length(&v.inline_inner);
+        remove(&mut v, 100);
+        remove(&mut v, 90);
+        remove(&mut v, 80);
+        remove(&mut v, 70);
+        remove(&mut v, 60);
+        remove(&mut v, 50);
+        remove(&mut v, 40);
+        remove(&mut v, 30);
+        remove(&mut v, 20);
+        assert!(vector::length(&v.inline_inner) == inline_len, 0);
+        remove(&mut v, 10);
+        assert!(vector::length(&v.inline_inner) + 1 == inline_len, 0);
+        remove(&mut v, 0);
+        assert!(vector::length(&v.inline_inner) + 2 == inline_len, 0);
+        assert!(length(&v) == 90, 0);
+
+        let index = 0;
+        i = 0;
+        while (i < 101) {
+            if (i % 10 != 0) {
+                assert!(*borrow(&v, index) == i, 0);
+                index = index + 1;
+            };
+            i = i + 1;
+        };
+        destroy(v);
+    }
+
+    #[test]
+    fun smart_vector_reverse_test() {
+        let v = empty();
+        let i = 0u64;
+        while (i < 10) {
+            push_back(&mut v, i);
+            i = i + 1;
+        };
+        reverse(&mut v);
+        let k = 0;
+        while (k < 10) {
+            assert!(*vector::borrow(&v.inline_inner, k) == 9 - k, 0);
+            k = k + 1;
+        };
+        while (i < 100) {
+            push_back(&mut v, i);
+            i = i + 1;
+        };
+        while (!vector::is_empty(&v.inline_inner)) {
+            remove(&mut v, 0);
+        };
+        reverse(&mut v);
+        i = 0;
+        let len = length(&v);
+        while (i + 1 < len) {
+            assert!(*big_vector::borrow(vector::borrow(&v.table_inner, 0), i) == *big_vector::borrow(vector::borrow(&v.table_inner, 0), i + 1) + 1, 0);
+            i = i + 1;
+        };
+        destroy(v);
+    }
+
+    #[test]
+    fun smart_vector_swap_test() {
+        let v = empty();
+        let i = 0;
+        while (i < 101) {
+            push_back(&mut v, i);
+            i = i + 1;
+        };
+        i = 0;
+        while (i < 51) {
+            swap(&mut v, i, 100 - i);
+            i = i + 1;
+        };
+        i = 0;
+        while (i < 101) {
+            assert!(*borrow(&v, i) == 100 - i, 0);
+            i = i + 1;
+        };
+        destroy(v);
+    }
+
+    #[test]
+    fun smart_vector_index_of_test() {
+        let v = empty();
+        let i = 0;
+        while (i < 100) {
+            push_back(&mut v, i);
+            i = i + 1;
+        };
+        i = 0;
+        while (i < 100) {
+            let (found, idx) = index_of(&mut v, &i);
+            assert!(found && idx == i, 0);
+            i = i + 1;
+        };
+        destroy(v);
+    }
+}
```

Comment:
Can we add more unit tests with edge cases?

Reply:
What are the edge cases in your mind?